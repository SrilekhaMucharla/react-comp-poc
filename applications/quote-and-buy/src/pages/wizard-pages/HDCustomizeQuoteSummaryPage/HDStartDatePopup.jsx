import { Col, Row } from 'react-bootstrap';
import { HDLabelRefactor } from 'hastings-components';
import React from 'react';
import PropTypes from 'prop-types';
import * as messages from './HDStartDatePopup.messages';
import {
    AnalyticsHDOverlayPopup as HDOverlayPopup
} from '../../../web-analytics';
// import './HDStartDatePopup.scss';
import InfoIcon from './svg/InfoIcon';

const HDStartDatePopup = (props) => {
    const { pageMetadata, className } = props;

    return (
        <HDOverlayPopup
            className={className}
            webAnalyticsView={{ ...pageMetadata, page_section: `${messages.customizeQuote} - ${messages.mainHeader}` }}
            webAnalyticsEvent={{ event_action: `${messages.customizeQuote} - ${messages.mainHeader}` }}
            id="popup-start-date"
            overlayButtonIcon={<InfoIcon width="20" height="20" color="#0085ff" />}
        >
            <Row className="popup-start-date">
                <Col>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                className="popup-start-date__label"
                                Tag="h2"
                                text={messages.mainHeader}
                                size="lg" />
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                className="popup-start-date__description"
                                text={messages.paragraphOne}
                                Tag="p" />
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <HDLabelRefactor
                                className="popup-start-date__description"
                                Tag="p"
                                text={messages.paragraphTwo} />
                        </Col>
                    </Row>
                </Col>
            </Row>
        </HDOverlayPopup>
    );
};

HDStartDatePopup.propTypes = {
    pageMetadata: PropTypes.shape({
        page_name: PropTypes.string.isRequired,
        page_type: PropTypes.string.isRequired,
        sales_journey_type: PropTypes.string.isRequired
    }).isRequired,
    className: PropTypes.string
};

HDStartDatePopup.defaultProps = {
    className: null
};

export default HDStartDatePopup;
