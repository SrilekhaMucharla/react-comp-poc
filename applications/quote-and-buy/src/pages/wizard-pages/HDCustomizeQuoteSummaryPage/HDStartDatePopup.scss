//TODO: check font family
.popup-start-date {
    color: $app-theme;
}

.popup-start-date__label {
    margin-top: 20px;
    line-height: 24px;
}

.popup-start-date__description {
    margin-top: 20px;
    font-weight: 500;
    font-size: $font-size-md;
    line-height: 24px;
    color: $info-color;
}

.popup-start-date > span {
    .label-text {
        color: $app-theme;
    }
}
