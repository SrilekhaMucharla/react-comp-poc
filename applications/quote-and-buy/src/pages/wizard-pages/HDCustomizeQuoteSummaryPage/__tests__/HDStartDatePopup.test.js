import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDStartDatePopup from '../HDStartDatePopup';

Enzyme.configure({ adapter: new Adapter() });

describe('<HDStartDatePopup />', () => {
    test('render component', () => {
        const wrapper = shallow((
            <HDStartDatePopup />
        ));
        expect(wrapper).toMatchSnapshot();
    });
});
