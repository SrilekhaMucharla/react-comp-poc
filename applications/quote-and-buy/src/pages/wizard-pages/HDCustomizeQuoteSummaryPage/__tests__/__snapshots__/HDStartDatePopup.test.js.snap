// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDStartDatePopup /> render component 1`] = `
<ForwardRef
  className={null}
  id="popup-start-date"
  onBeforeOpen={null}
  overlayButtonIcon={
    <InfoIcon
      color="#0085ff"
      height="20"
      width="20"
    />
  }
  webAnalyticsEvent={
    Object {
      "event_action": "Customise quote - Your start date",
    }
  }
  webAnalyticsView={
    Object {
      "page_section": "Customise quote - Your start date",
    }
  }
>
  <Row
    className="popup-start-date"
    noGutters={false}
  >
    <Col>
      <Row
        noGutters={false}
      >
        <Col>
          <HDLabelRefactor
            Tag="h2"
            adjustImagePosition={true}
            className="popup-start-date__label"
            size="lg"
            text="Your start date"
          />
        </Col>
      </Row>
      <Row
        noGutters={false}
      >
        <Col>
          <HDLabelRefactor
            Tag="p"
            adjustImagePosition={true}
            className="popup-start-date__description"
            text="Your start date is when you need the policy to begin. This will be either to continue insuring a car from the date the existing cover is due to end or to start insuring a new car."
          />
        </Col>
      </Row>
      <Row
        noGutters={false}
      >
        <Col>
          <HDLabelRefactor
            Tag="p"
            adjustImagePosition={true}
            className="popup-start-date__description"
            text="All cover will start at 00:01 on your chosen start date, unless you want your cover to start today (in which case it will start as soon as you buy the policy)."
          />
        </Col>
      </Row>
    </Col>
  </Row>
</ForwardRef>
`;
